content/articles/integrate-airtable-with-nextjs/index.md

```diff
@@ -0,0 +1,601 @@
+
+export default async (req, res) => {
+	const { id } = req.body
+	try {
+		const deletedRecords = await table.destroy([id])
+		res.status(200).json(deletedRecords)
+	} catch (error) {
+		console.log(error)
+		res.status(500).json({ msg: 'Something went wrong! 😕' })
+	}
+}
+```
+
+### Creating front-end
+Now we have our API ready for all our CRUD operations. Let's create the interface to display this data in our Next.js application. Head over to the `index.js` file in your `/pages` directory and get rid of all the code in it. Add the following code:
+
+```jsx
+import Head from 'next/head'
+
+export default function Home() {
+	return (
+		<div className='container mx-auto my-6 max-w-xl'>
+			<Head>
+				<title>@Grocery List</title>
+			</Head>
+
+			<main>
+				<p className='text-2xl font-bold text-grey-800 py-2'>🛒 Grocery List</p>
+			</main>
+		</div>
+	)
+}
+
+```
+
+The above code is nothing but a React functional component. Next.js has a built-in `Head` component that will act as the `head` tag of your HTML page. Now, if you start running the server, you will see your app. Tailwindcss is a class-based framework. You have to add the classes according to the styles you want, just like in [bootstrap](https://getbootstrap.com).
+
+Next.js has a built-in function `getServerSideProps` for enabling the server-side rendering inside a page. Next.js executes the code inside this function every time before rendering the page. Learn more on `getServersideProps` [here](https://nextjs.org/docs/basic-features/data-fetching###getserversideprops-server-side-rendering).
+
+We will fetch all the items from the Airtable and then pass them to the `Home` component as [`props`](https://reactjs.org/docs/components-and-props.html) to use the data. Add the following function to your `index.js` page.
+
+```jsx
+import { table, minifyItems } from '../utils/Airtable'
+
+export default function Home({ initialItems }) {
+	console.log(initialItems)
+	// ...
+	}
+
+export async function getServerSideProps(context) {
+	try {
+		const items = await table.select({}).firstPage()
+		return {
+			props: {
+				initialItems: minifyItems(items),
+			},
+		}
+	} catch (error) {
+		console.log(error)
+		return {
+			props: {
+				err: 'Something went wrong 😕',
+			},
+		}
+	}
+}
+
+```
+
+The above function gets all the records from the Airtable and passes them to `initialItems`. For now, see if it's working by just console logging the data.
+
+### React Context API to Integrate Airtable Data
+> Context provides a way to pass data through the component tree passing props down manually at every level - [React documentation](https://reactjs.org/docs/context.html).
+
+In large-scale projects, we have to use the data in many components. So it is a better idea to use React context instead of passing props. Create a new context folder and add a new `items.js` file. Here, we will perform all the operations on Airtable data and pass the data to the front-end.
```

Comment:
```suggestion
In large-scale projects, we have to use the data in many components. So it is a better idea to use React context instead of passing props. Create a new `context` folder and add a new `items.js` file. Here, we will perform all the operations on Airtable data and pass the data to the front-end.
```